### PR TITLE
[Android] Implement logic to move focus to next Entry 

### LINKED ISSUE #5724: Entry field Enter does not always move to next field
### Additional Notes
- Good Summary of behavior:  https://github.com/dotnet/maui/issues/5724#issuecomment-1165282553
- `XF` implemented the forward motion [manually](https://github.com/xamarin/Xamarin.Forms/blob/d7d5e01a7435f2fd055485959df8708c03911f67/Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs#L92). On `Android` the forward behavior happens automatically so I'm curious why this behavior doesn't work automatically. Ideally, we can just make it work opposed to having to implement our own focus search code. 
  - It looks like `Next` fails to work because for `OnEditorAction` we set `e.Handled` to true for everything so `Android` doesn't fall back to its default behavior ever

### Description

On a screen with multiple entry fields the treatment of the enter key is inconsistent. What I've seen so far:

On Android, if `ReturnType` is set to `Next` the on screen key appearance changes to a 'Tab' icon and clicking it does nothing. If `ReturnType="Done"` the icon changes to an "Enter" icon and clicking it calls the Completed action. In neither case does it advance focus to the next field. 

On Windows pressing the enter key calls `OnCompleted` whether `ReturnType` is set to `Done` or `Next` and the the focus switches to the next field if `ReturnType` is set to `Next` (and not `Done`) all of which seems correct.

Xamarin Forms acts like Windows (above) except that pressing enter in a field where `ReturnType="Done"` causes the field to be unfocused rather than focusing on the next field as `Next` would.

### Steps to Reproduce

1. Unzip the project in [Simple-Entry.zip](https://github.com/dotnet/maui/files/8393572/Simple-Entry.zip)
2. Compile and run the program on Android
3. Select the first field
4. Click on the on screen keyboard button to advance to the next field
5. Note that nothing happens (no call of OnCompleted, no change in focus)

Here's what you should see (the blue virtual key is the one you should click on):
 
Simple-Entry

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11, Windows 10 

### Did you find any workaround?

No

### Relevant log output

_No response_

### LINKED ISSUE #10858: [Android] Keyboard is not dismissed after tapping "Enter"
### Description

When editing an Entry, the virtual keyboard is not dismissed after tapping the "Enter" button:
maui_entry_bug
This behaviour does not occur on iOS.

### Steps to Reproduce

1. Create a new .NET MAUI project
2. Add an entry
3. Tap into the entry control, type some text and tap "Enter"
4. The keyboard does not disappear, it remains visible until the back button is pressed

### Link to public reproduction project repository

https://github.com/owl-guy/EditorExample

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11 and up

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Implement logic to move focus to next Entry on Android.

fix-5724

### Issues Fixed

Fixes #5724 
Fixes #10858

### CODE REVIEW

File: src/Core/src/Platform/Android/FocusExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using Android.Runtime;
+using Android.Views;
+using AView = Android.Views.View;
+
+namespace Microsoft.Maui.Controls.Platform
+{
+	public static class FocusExtensions
+	{
+		public static AView? FocusSearch(this AView focused, [GeneratedEnum] FocusSearchDirection direction)
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Android/FocusExtensions.cs:9]: This implementation is simple, we use `FocusFinder` to move the focus to the next element. In Xamarin.Forms we had https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.iOS/VisualElementRenderer.cs#L194

Where we looked for the element with the next index using the `ITabStopElement `property. But, now we do not have this property. Would this be correct, or can we do something better?

Comment (rachelkang) [reply src/Core/src/Platform/Android/FocusExtensions.cs:9]: We woud need to check what the screen reader behavior is - does the TalkBack focus move with the keyboard focus? If it's inconsistent, we need to figure out what the desired behavior is there. If it's consistent, then I think this is an OK solution

Comment (PureWeen) [reply src/Core/src/Platform/Android/FocusExtensions.cs:9]: We shouldn't copy this code from XF. 100 percent of the code in XF that influenced focus is wrong and was most likely created to work around fundamental flaws of things that XF made clickable and focusable that stacks of hacks had to be built against.

Comment (rachelkang) [reply src/Core/src/Platform/Android/FocusExtensions.cs:9]: sorry, I misread which APIs were being used - taking back what I said and seconding Shane here!

Comment (PureWeen) [reply src/Core/src/Platform/Android/FocusExtensions.cs:9]: If we want to enable this behavior then we should do it through other means like a behavior "https://github.com/CommunityToolkit/Maui/issues/36"

Or through additional APIs so it's opt in

**Review by PureWeen — CHANGES_REQUESTED** [review]
I think we should figure out why .NET MAUI by default doesn't behave the same as a default Android app. 

If you load up a default android app using net6.0-android the enter key moves between the fields where as with .NET MAUI it doesn't

Why is there a difference? Ideally we can just lean on the default platform behavior instead of needing to figure it out for ourselves.

**Comment by pulmuone** [comment]
In MAUI.iOS, it works the same as in Xamarin.Forms, but not in Maui.Android.

**Comment by jsuarezruiz** [comment]
To help focus and move faster, we're going to have fewer open PRs and focus on the ones we'll be working on in each release. This PR is not included in the next release so I close it for now. I will reopen it and adapt to changes later.

**Comment by owl-guy** [comment]
@jsuarezruiz This feature should have been a fix for https://github.com/dotnet/maui/issues/10858, so what about that? The keyboard not being dismissed on Android is a serious problem, as it can prevent apps from being published. Every app that has even remotely to do with filling in forms is unusable!

**Comment by angrynik** [comment]
Please look at this. Maui is almost becoming good, but I need basic UI functionality on Android.

**Comment by PureWeen** [comment]
> Please look at this. Maui is almost becoming good, but I need basic UI functionality on Android.

This is fixed on the latest version of net7

**Comment by angrynik** [comment]
I'm on .net7 latest 7.0.1.
I have a Maui App with some Entry controls on a ContentPage.
Does not move focus on tab from the virtual keyboard.

**Comment by MichaelShapiro** [comment]
Perhaps a stupid question - but why is this closed? The Entry with a ReturnType = "Next" still doesn't advance to the next Entry field. The only thing that has changed is that now the Completed event finally fires (before it only fired for ReturnType = "Done")

**Comment by rachelkang** [comment]
Hi folks! This issue was closed as it has been fixed. While you aren't seeing the fix yet, you will be able to see the fix in the next release. Thank you for your patience :)

**Pull request closed without merging**